Dockerfile.debug

```diff
@@ -1,90 +1,108 @@
+COPY dashboard/.yarnrc dashboard/.eslintrc.json dashboard/next.config.js dashboard/postcss.config.js dashboard/tailwind.config.js dashboard/tsconfig.json dashboard/tslint.json ./
 COPY dashboard/public public
 COPY dashboard/src src
 
-## Install stage dependencies
-COPY dashboard/package.json dashboard/yarn.lock ./
-RUN yarn install
+## Build
+RUN yarn build
 
 #####################################################################################################################################################
 
 # Engine
-FROM base AS engine-stage
+FROM build AS engine-stage
 
 WORKDIR /engine
 
+## Install stage dependencies
+COPY .yarnrc package.json yarn.lock ./
+RUN yarn install --frozen-lockfile
+
 ## Copy files
 COPY ormconfig.js tsconfig.json ./
 COPY src src
 
-## Install stage dependencies
-COPY package.json yarn.lock ./
-RUN yarn install
-
-## Build
-RUN yarn build
-
 #####################################################################################################################################################
 
 # Main stage
 FROM base AS main-stage
 
-## Copy from dashboard-stage
-WORKDIR /dashboard
-COPY --from=dashboard-stage /dashboard/*.* ./
-COPY --from=dashboard-stage /dashboard/src src
-COPY --from=dashboard-stage /dashboard/public public
-COPY --from=dashboard-stage /dashboard/node_modules node_modules
-
-## Copy from engine-stage
-WORKDIR /engine
-COPY --from=engine-stage /engine/ormconfig.js /engine/tsconfig.json /engine/package.json ./
-COPY --from=engine-stage /engine/node_modules node_modules
-COPY --from=engine-stage /engine/dist dist
+## Copy from pgsql-stage
+WORKDIR /
+COPY --from=pgsql-stage /usr/lib/postgresql /usr/lib/postgresql
+COPY --from=pgsql-stage /usr/share/postgresql /usr/share/postgresql
 
 ## Copy files
 COPY ./src/scripts/postgresql.conf /etc/postgresql/14/main/postgresql.conf
 COPY ./src/scripts/pg_hba.conf /etc/postgresql/14/main/pg_hba.conf
-COPY docker-entrypoint-dev.sh /engine/docker-entrypoint-dev.sh
+COPY docker-entrypoint.debug.sh /engine/docker-entrypoint.debug.sh
 COPY src/scripts /engine/src/scripts
 
+## Copy from engine-stage
+WORKDIR /engine
+COPY --from=engine-stage /engine/node_modules node_modules
+COPY --from=engine-stage /engine/package.json /engine/ormconfig.js /engine/tsconfig.json ./
+COPY --from=engine-stage /engine/src src
+
+## Copy from dashboard-stage
+WORKDIR /dashboard
+COPY --from=dashboard-stage /dashboard/public ./public
+COPY --from=dashboard-stage /dashboard/.next/standalone ./
+COPY --from=dashboard-stage /dashboard/.next/static ./.next/static
+
 ## Default ENVs that can be overwritten
-ARG IASQL_ENV=dev
+ARG IASQL_ENV=debug
 ENV IASQL_ENV=$IASQL_ENV
-ENV NEXT_PUBLIC_IASQL_ENV=$IASQL_ENV
 ARG IASQL_TELEMETRY=off
 ENV IASQL_TELEMETRY=$IASQL_TELEMETRY
-ENV NEXT_PUBLIC_IASQL_TELEMETRY=$IASQL_TELEMETRY
 ARG DB_USER=postgres
```

Comment:
without `NEXT_PUBLIC_` variables, does the container work?

Reply:
yes! it is the same configuration as the original `Dockerfile` just a few tweaks for the engine to enable the debug and hot-reload.